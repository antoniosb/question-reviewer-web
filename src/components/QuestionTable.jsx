
import React from 'react'
import QuestionTableItem from './QuestionTableItem'
import QuestionTableItemHeader from './QuestionTableItemHeader'
import { Collapse, Icon, Alert } from 'antd'
import '../assets/scss/components/question-table.scss'
const Panel = Collapse.Panel

class QuestionTable extends React.PureComponent {
  render() {
    if (this.props.value.length < 1) {
      return (<Alert message="Nenhuma questÃ£o encontrada" type="info" />)
    }
    return (
      <Collapse accordion className="question-table">
        { this.props.value.map((item, idx) => 
          <Panel header={<QuestionTableItemHeader item={item} />} key={idx} className="question-table-item">
            <QuestionTableItem value={item} />
          </Panel>
         )
        }
      </Collapse>
    )
  }
}

export default QuestionTable