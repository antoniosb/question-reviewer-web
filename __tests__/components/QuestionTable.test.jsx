import React from 'react'
import QuestionTable from '../../src/components/QuestionTable'

describe('Components > QuestionTable', () => {
  test('Render', () => {
    const component = shallow(<QuestionTable value={[{
      user: { login: 'teste' },
      id: 1,
      updated_at: new Date(2000, 11, 1, 12, 1, 1).toISOString(),
      question_alternatives: [{},{},{},{},{}]
    }]} />)

    expect(component).toMatchSnapshot()
  })
  test('Render Not Found', () => {
    const component = shallow(<QuestionTable value={[]} />)

    expect(component).toMatchSnapshot()
  })
})