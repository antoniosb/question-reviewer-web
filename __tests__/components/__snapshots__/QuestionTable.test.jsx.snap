// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > QuestionTable Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionTable
    value={
      Array [
        Object {
          "id": 1,
          "question_alternatives": Array [
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
          ],
          "updated_at": "2000-12-01T14:01:01.000Z",
          "user": Object {
            "login": "teste",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accordion": true,
      "bordered": true,
      "children": Array [
        <CollapsePanel
          className="question-table-item"
          header={
            <QuestionTableItemHeader
              item={
                Object {
                  "id": 1,
                  "question_alternatives": Array [
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                  ],
                  "updated_at": "2000-12-01T14:01:01.000Z",
                  "user": Object {
                    "login": "teste",
                  },
                }
              }
            />
          }
        >
          <withRouter(Connect(QuestionTableItem))
            value={
              Object {
                "id": 1,
                "question_alternatives": Array [
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                "updated_at": "2000-12-01T14:01:01.000Z",
                "user": Object {
                  "login": "teste",
                },
              }
            }
          />
        </CollapsePanel>,
      ],
      "className": "question-table",
      "openAnimation": Object {
        "appear": [Function],
        "enter": [Function],
        "leave": [Function],
      },
      "prefixCls": "ant-collapse",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": <withRouter(Connect(QuestionTableItem))
            value={
              Object {
                "id": 1,
                "question_alternatives": Array [
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                "updated_at": "2000-12-01T14:01:01.000Z",
                "user": Object {
                  "login": "teste",
                },
              }
            }
          />,
          "className": "question-table-item",
          "header": <QuestionTableItemHeader
            item={
              Object {
                "id": 1,
                "question_alternatives": Array [
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                "updated_at": "2000-12-01T14:01:01.000Z",
                "user": Object {
                  "login": "teste",
                },
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "value": Object {
              "id": 1,
              "question_alternatives": Array [
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
              ],
              "updated_at": "2000-12-01T14:01:01.000Z",
              "user": Object {
                "login": "teste",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accordion": true,
        "bordered": true,
        "children": Array [
          <CollapsePanel
            className="question-table-item"
            header={
              <QuestionTableItemHeader
                item={
                  Object {
                    "id": 1,
                    "question_alternatives": Array [
                      Object {},
                      Object {},
                      Object {},
                      Object {},
                      Object {},
                    ],
                    "updated_at": "2000-12-01T14:01:01.000Z",
                    "user": Object {
                      "login": "teste",
                    },
                  }
                }
              />
            }
          >
            <withRouter(Connect(QuestionTableItem))
              value={
                Object {
                  "id": 1,
                  "question_alternatives": Array [
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                  ],
                  "updated_at": "2000-12-01T14:01:01.000Z",
                  "user": Object {
                    "login": "teste",
                  },
                }
              }
            />
          </CollapsePanel>,
        ],
        "className": "question-table",
        "openAnimation": Object {
          "appear": [Function],
          "enter": [Function],
          "leave": [Function],
        },
        "prefixCls": "ant-collapse",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": <withRouter(Connect(QuestionTableItem))
              value={
                Object {
                  "id": 1,
                  "question_alternatives": Array [
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                  ],
                  "updated_at": "2000-12-01T14:01:01.000Z",
                  "user": Object {
                    "login": "teste",
                  },
                }
              }
            />,
            "className": "question-table-item",
            "header": <QuestionTableItemHeader
              item={
                Object {
                  "id": 1,
                  "question_alternatives": Array [
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                    Object {},
                  ],
                  "updated_at": "2000-12-01T14:01:01.000Z",
                  "user": Object {
                    "login": "teste",
                  },
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "value": Object {
                "id": 1,
                "question_alternatives": Array [
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                "updated_at": "2000-12-01T14:01:01.000Z",
                "user": Object {
                  "login": "teste",
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Components > QuestionTable Render Not Found 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionTable
    value={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "message": "Nenhuma questão encontrada",
      "type": "info",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "message": "Nenhuma questão encontrada",
        "type": "info",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
